src/app/Library/CrudPanel/Traits/Tabs.php

```diff
@@ -106,56 +106,94 @@ public function isLastTab($label)
+     * Instead, use $this->getElementsWithoutATab($this->getCurrentFields())
+     *
      * @return \Illuminate\Support\Collection
      */
     public function getFieldsWithoutATab()
     {
-        $all_fields = $this->getCurrentFields();
+        return $this->getElementsWithoutATab($this->getCurrentFields());
+    }
 
-        $fields_without_a_tab = collect($all_fields)->filter(function ($value, $key) {
+    /**
+     * @return \Illuminate\Support\Collection
+     */
+    public function getElementsWithoutATab(array $elements)
+    {
+        return collect($elements)->filter(function ($value) {
             return ! isset($value['tab']);
         });
+    }
 
-        return $fields_without_a_tab;
+    /**
+     * @deprecated Do not use this method as it will be removed in future versions!
+     * Instead, use $this->getTabItems($tabLabel, 'fields')
+     *
+     * @return array|\Illuminate\Support\Collection
+     */
+    public function getTabFields(string $tabLabel)
+    {
+        return $this->getTabItems($tabLabel, 'fields');
     }
 
     /**
-     * @param $label
      * @return array|\Illuminate\Support\Collection
      */
-    public function getTabFields($label)
+    public function getTabItems(string $tabLabel, string $source)
     {
-        if ($this->tabExists($label)) {
-            $all_fields = $this->getCurrentFields();
+        if (in_array($tabLabel, $this->getUniqueTabNames($source))) {
+            $items = $this->getCurrentItems($source);
 
-            $fields_for_current_tab = collect($all_fields)->filter(function ($value, $key) use ($label) {
-                return isset($value['tab']) && $value['tab'] == $label;
+            return collect($items)->filter(function ($value) use ($tabLabel) {
+                return isset($value['tab']) && $value['tab'] == $tabLabel;
             });
-
-            return $fields_for_current_tab;
         }
 
         return [];
     }
 
+    public function getTabs(): array
+    {
+        return $this->getUniqueTabNames('fields');
+    }
+
     /**
-     * @return array
+     * $source could be `fields` or `columns` for now.
      */
-    public function getTabs()
+    public function getUniqueTabNames(string $source): array
     {
         $tabs = [];
-        $fields = $this->getCurrentFields();
+        $items = $this->getCurrentItems($source);
 
-        $fields_with_tabs = collect($fields)
-            ->filter(function ($value, $key) {
+        collect($items)
+            ->filter(function ($value) {
                 return isset($value['tab']);
             })
-            ->each(function ($value, $key) use (&$tabs) {
+            ->each(function ($value) use (&$tabs) {
                 if (! in_array($value['tab'], $tabs)) {
                     $tabs[] = $value['tab'];
                 }
             });
 
         return $tabs;
     }
+
+    public function getCurrentItems(string $source): array
```

Comment:
It's only used in this file and has implementation details that we may want to change in future.
We don't want people to be using this method directly. 

```suggestion
    private function getCurrentItems(string $source): array
```